aws-datastore/src/test/java/com/amplifyframework/datastore/storage/InMemoryStorageAdapter.java

```diff
@@ -144,22 +159,23 @@ public void initialize(
             @NonNull final Consumer<StorageItemChange.Record> onSuccess,
             @NonNull final Consumer<DataStoreException> onError
     ) {
-        for (Model savedItem : items) {
-            if (savedItem.getId().equals(item.getId())
-                    && (predicate == null || predicate.evaluate(item))) {
-                items.remove(item);
-                StorageItemChange.Record deletion = StorageItemChange.<T>builder()
-                        .item((T) savedItem)
-                        .itemClass((Class<T>) savedItem.getClass())
-                        .type(StorageItemChange.Type.DELETE)
-                        .predicate(predicate)
-                        .initiator(initiator)
-                        .build()
-                        .toRecord(storageItemChangeConverter);
-                changeRecordStream.onNext(deletion);
-                onSuccess.accept(deletion);
-                return;
-            }
+        final int index = indexOf(item);
+        if (index > -1 && (predicate == null || predicate.evaluate(item))) {
```

Comment:
You could invert this expression, and put the error handling in the body.

Then, do your happy path back -4 indentation to the left.

e.g.
```
if (unlikeBadThing) {
    announce("Hey no fair...");
    return;
}
// otherwise
performBusiness();
asUsual();
etc();
```

Reply:
good idea. Also caught a bug where i checked the predicate condition against item to delete rather than on the already stored item.